test/functional/services/dashboard/add_panel.js

```diff
@@ -93,6 +93,11 @@ export function DashboardAddPanelProvider({ getService, getPageObjects }) {
       }
     }
 
+    async waitForEuiTableLoading() {
```

Comment:
Seems like this function could go somewhere more general as it might be useful for other places that use the eui table and want to wait for it to load.  I don't see a good spot for it right now.  Maybe a new service named `euiComponents`?  Could take in an optional selector name to use to narrow down but I wonder if you can use it without an element, directly on remote (just like we do for the global loading indicator - making sure it doesn't exist at all, anywhere on the page).

Reply:
I thought about that.  But the euiBasicTable can get populated from any data source and doesn't necessarily require a request/response to Elasticsearch so I'm not sure what to do.

Again, the retry loop in setValue seems like it should always handle the case where the field isn't initially user-editable.

Reply:
> Seems like this function could go somewhere more general as it might be useful for other places that use the eui table and want to wait for it to load. I don't see a good spot for it right now. Maybe a new service named euiComponents?

There is a flyout service and I have a PR out for a combo box service. https://github.com/elastic/kibana/pull/21219.